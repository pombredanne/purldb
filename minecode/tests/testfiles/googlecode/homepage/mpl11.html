

<!DOCTYPE html>
<html>
<head>
 <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" >
 <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1" >
 
 <meta name="ROBOTS" content="NOARCHIVE">
 
 <link rel="icon" type="image/vnd.microsoft.icon" href="http://www.gstatic.com/codesite/ph/images/phosting.ico">
 
 
 <script type="text/javascript">
 
 
 
 
 var codesite_token = null;
 
 
 var CS_env = {"assetHostPath": "http://www.gstatic.com/codesite/ph", "profileUrl": null, "projectName": "gpsee", "relativeBaseUrl": "", "loggedInUserEmail": null, "projectHomeUrl": "/p/gpsee", "domainName": null, "token": null, "assetVersionPath": "http://www.gstatic.com/codesite/ph/11512667522589661435"};
 var _gaq = _gaq || [];
 _gaq.push(
 ['siteTracker._setAccount', 'UA-18071-1'],
 ['siteTracker._trackPageview']);
 
 _gaq.push(
 ['projectTracker._setAccount', 'UA-10000054-1'],
 ['projectTracker._trackPageview']);
 
 (function() {
 var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
 ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
 (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(ga);
 })();
 
 </script>
 
 
 <title>
 gpsee -
 
 
 General Purpose Spidermonkey Embedding Ecosystem - Google Project Hosting
 </title>
 <link type="text/css" rel="stylesheet" href="http://www.gstatic.com/codesite/ph/11512667522589661435/css/core.css">
 
 <link type="text/css" rel="stylesheet" href="http://www.gstatic.com/codesite/ph/11512667522589661435/css/ph_detail.css" >
 
 
 
 
<!--[if IE]>
 <link type="text/css" rel="stylesheet" href="http://www.gstatic.com/codesite/ph/11512667522589661435/css/d_ie.css" >
<![endif]-->
 <style type="text/css">
 .menuIcon.off { background: no-repeat url(http://www.gstatic.com/codesite/ph/images/dropdown_sprite.gif) 0 -42px }
 .menuIcon.on { background: no-repeat url(http://www.gstatic.com/codesite/ph/images/dropdown_sprite.gif) 0 -28px }
 .menuIcon.down { background: no-repeat url(http://www.gstatic.com/codesite/ph/images/dropdown_sprite.gif) 0 0; }
 
 
 .psicon-container {
 min-width:24px;
 }

 
 </style>
</head>
<body class="t1">
<script type="text/javascript">
 window.___gcfg = {lang: 'en'};
 (function() 
 {var po = document.createElement("script");
 po.type = "text/javascript"; po.async = true;po.src = "https://apis.google.com/js/plusone.js";
 var s = document.getElementsByTagName("script")[0];
 s.parentNode.insertBefore(po, s);
 })();
</script>
<div class="headbg">

 <div id="gaia">
 

 <span>
 
 
 <a href="#" id="projects-dropdown" onclick="return false;"><u>My favorites</u> <small>&#9660;</small></a>
 | <a href="https://www.google.com/accounts/ServiceLogin?service=code&amp;ltmpl=phosting&amp;continue=http%3A%2F%2Fcode.google.com%2Fp%2Fgpsee%2F&amp;followup=http%3A%2F%2Fcode.google.com%2Fp%2Fgpsee%2F" onclick="_CS_click('/gb/ph/signin');"><u>Sign in</u></a>
 
 </span>

 </div>

 <div class="gbh" style="left: 0pt;"></div>
 <div class="gbh" style="right: 0pt;"></div>
 
 
 <div style="height: 1px"></div>
<!--[if lte IE 7]>
<div style="text-align:center;">
Your version of Internet Explorer is not supported. Try a browser that
contributes to open source, such as <a href="http://www.firefox.com">Firefox</a>,
<a href="http://www.google.com/chrome">Google Chrome</a>, or
<a href="http://code.google.com/chrome/chromeframe/">Google Chrome Frame</a>.
</div>
<![endif]-->



 <table style="padding:0px; margin: 0px 0px 10px 0px; width:100%" cellpadding="0" cellspacing="0"
 itemscope itemtype="http://schema.org/CreativeWork">
 <tr style="height: 58px;">
 
 
 
 <td id="plogo">
 <link itemprop="url" href="/p/gpsee">
 <a href="/p/gpsee/">
 
 
 <img src="/p/gpsee/logo?cct=1356033783"
 alt="Logo" itemprop="image">
 
 </a>
 </td>
 
 <td style="padding-left: 0.5em">
 
 <div id="pname">
 <a href="/p/gpsee/"><span itemprop="name">gpsee</span></a>
 </div>
 
 <div id="psum">
 <a id="project_summary_link"
 href="/p/gpsee/"><span itemprop="description">General Purpose Spidermonkey Embedding Ecosystem</span></a>
 
 </div>
 
 
 </td>
 <td style="white-space:nowrap;text-align:right; vertical-align:bottom;">
 
 <form action="/hosting/search">
 <input size="30" name="q" value="" type="text">
 
 <input type="submit" name="projectsearch" value="Search projects" >
 </form>
 
 </tr>
 </table>

</div>

 
<div id="mt" class="gtb"> 
 <a href="/p/gpsee/" class="tab active">Project&nbsp;Home</a>
 
 
 
 
 
 
 <a href="/p/gpsee/w/list" class="tab ">Wiki</a>
 
 
 
 
 
 <a href="/p/gpsee/issues/list"
 class="tab ">Issues</a>
 
 
 
 
 
 <a href="/p/gpsee/source/checkout"
 class="tab ">Source</a>
 
 
 
 
 
 
 
 
 <div class=gtbc></div>
</div>
<table cellspacing="0" cellpadding="0" width="100%" align="center" border="0" class="st">
 <tr>
 
 
 
 
 
 
 
 <td class="subt">
 <div class="st1">
 <div class="isf">
 <span class="inst1">
 <a href="/p/gpsee/">Summary</a>
 </span>
 
 
 &nbsp;
 <span class="inst3">
 <a href="/p/gpsee/people/list">People</a>
 </span>
 
 
 </div>
</div>

 </td>
 
 
 <td align="right" valign="top" class="bevel-right"></td>
 </tr>
</table>


<script type="text/javascript">
 var cancelBubble = false;
 function _go(url) { document.location = url; }
</script>
<div id="maincol"
 
>

 





<table width="100%">
 <tr class="pscontent">
 <td class="pscolumnl">
 
 <div class="phead">Project Information</div>
 
 
 
 <ul class="pslist">
 
 <li class="psmeta">
 <a href="/p/gpsee/feeds">Project feeds</a>
 </li>
 <li class="psgap"/>
 
 
 
 <li class="psmeta"><b>Code license</b></li>
 <li class="psmeta">
 <a href="http://www.mozilla.org/MPL/" rel="nofollow">Mozilla Public License 1.1</a>
 
 </li>
 <li class="psgap"/>
 
 
 
 
 
 
 <li class="psmeta">
 <span id="project_labels">
 <b>Labels</b><br/>
 
 <a class="label"
 href="/hosting/search?q=label:CommonJS">CommonJS</a>, 
 
 <a class="label"
 href="/hosting/search?q=label:interoperablejs">interoperablejs</a>, 
 
 <a class="label"
 href="/hosting/search?q=label:javascript">javascript</a>, 
 
 <a class="label"
 href="/hosting/search?q=label:spidermonkey">spidermonkey</a>, 
 
 <a class="label"
 href="/hosting/search?q=label:ssjs">ssjs</a>, 
 
 <a class="label"
 href="/hosting/search?q=label:serverjs">serverjs</a>
 
 </span>
 </li>
 <li class="psgap"/>
 
 
 </ul>
 <div class="psicon">
 <div class="psicon-container goog-inline-block">
 <div style="float:right" class="SPRITE_people-y16 goog-inline-block vt"></div>
 </div>
 <span><b>Members</b></span>
 </div>
 <ul class="pslist">
 
 
 
 

 <a class="userlink"
 href="/u/108506978699427090444/">w...@page.ca</a
>, 
 
 

 <a class="userlink"
 href="/u/wes.page.ca@gmail.com/">wes.page.ca@gmail.com</a
>, 
 
 

 <a class="userlink"
 href="/u/111470924408745629945/">wes%page...@gtempaccount.com</a
>
 
 
 
 <li class="psmeta">
 <a href="people/list">8 committers</a>
 </li>
 
 
 <li class="psmeta">
 <a href="people/list">1 contributor</a>
 </li>
 
 <li class="psgap"/>
 
 
 </ul>
 
 
 
 
 <div class="phead">Links</div>
 <ul class="pslist" style="white-space:nowrap">
 
 <li class="psmeta"><b>Blogs</b></li>
 
 <li class="psmeta">
 <a href="http://gpsee.blogspot.com" rel="nofollow">GPSEE Blog</a>
 </li>
 
 <li class="psgap"/>
 
 
 <li class="psmeta"><b>External links</b></li>
 
 <li class="psmeta">
 <a href="http://code.google.com/p/gpsee/wiki/Building" rel="nofollow">Building GPSEE</a>
 </li>
 
 <li class="psmeta">
 <a href="http://www.page.ca/~wes/opensource/gpsee/" rel="nofollow">GPSEE Reference Docs</a>
 </li>
 
 <li class="psgap"/>
 
 
 <li class="psmeta"><b>Groups</b></li>
 
 <li class="psmeta">
 <a href="http://groups.google.com/group/gpsee" rel="nofollow">GPSEE Google Group</a>
 </li>
 
 <li class="psgap"/>
 
 </ul>
 
 </td>
 <td id="wikicontent" class="psdescription">
 <h1><a name="What_is_GPSEE?"></a>What is GPSEE?<a href="#What_is_GPSEE?" class="section_anchor"></a></h1><ul><li>a platform for developing and running <a href="/p/gpsee/wiki/CommonJS">CommonJS</a> programs </li><li>a general-purpose C API for embedding <a href="/p/gpsee/wiki/SpiderMonkey">SpiderMonkey</a> + CommonJS </li><li>a general-purpose C API for adding interoperability between JSAPI projects </li><li>licensed under the exact same terms as <a href="/p/gpsee/wiki/SpiderMonkey">SpiderMonkey</a> (MPL 1.1, GPLv2, LGPL 2.1) </li><li>pronounced &quot;<i>gypsy</i>&quot; </li></ul><p>  GPSEE is written and maintained by <a href="/p/gpsee/wiki/PageMail">PageMail</a>, Inc. The primary developers are Wes Garland and Donny Viszneki, however, we are actively looking for contributions from the community! GPSEE is an offshoot of earlier, proprietary research and development at <a href="/p/gpsee/wiki/PageMail">PageMail</a>. That project&#x27;s primary goal was to create an easy-to-embed <a href="/p/gpsee/wiki/JavaScript">JavaScript</a> language interpreter which can be used as glue and configuration logic in large software systems, with interest in the ability to write batch jobs and serve web content. </p><p>GPSEE is delivered as one unit, but compromises three main  facets: </p><ol><li><strong>gsr</strong>, the GPSEE Script Runner, which executes CommonJS programs, including via shell-script &quot;shebang&quot; facilities. GPSEE also ships with some sample REPLs (shells), including a CommonJS-enabled version of Mozilla&#x27;s JS Shell. </li><li><strong>GPSEE-core</strong> is a group of facilities which allow embedders to load modules, and to use <a href="/p/gpsee/wiki/SpiderMonkey">SpiderMonkey</a>&#x27;s limited-use facilities (e.g. context private storage) without worrying about stepping on other modules&#x27; toes. GPSEE-core is designed to be call-compatible with <a href="/p/gpsee/wiki/SpiderMonkey">SpiderMonkey</a>, so that it may be turned on and off with a simple header file change. </li><li><strong>modules</strong> is a directory with &quot;stock&quot; modules, adding functionality to the <a href="/p/gpsee/wiki/JavaScript">JavaScript</a> language such as ByteString, ByteArray, POSIX signals, CGI, and FFI (foreign-function interface). The FFI module allows painless wrapping of C functions and datatypes for use in creating new <a href="/p/gpsee/wiki/JavaScript">JavaScript</a> modules. </li></ol><h1><a name="POSIX"></a>POSIX<a href="#POSIX" class="section_anchor"></a></h1><p>GPSEE&#x27;s FFI module is specifically targeted at reflecting POSIX into <a href="/p/gpsee/wiki/JavaScript">JavaScript</a> -- in a way which does not require runtime detection of the underlying architecture or other portability hacks.  GPSEE targets the SUSv3 and SVID interface groups. </p><p>GPSEE FFI code which uses POSIX functions will run the same on every operating system and CPU, regardless of the underlying implementation. The <a href="/p/gpsee/wiki/JavaScript">JavaScript</a> programmer does not need to worry about details like struct layout, pointer sizes, or even whether the interface is defined as a C function or a C pre-processor macro. </p><h1><a name="Support"></a>Support<a href="#Support" class="section_anchor"></a></h1><p>GPSEE is provided as-is, without warranty of any kind, express or implied, including but not limited to the warranties of merchantability, or fitness for any purpose. We do not guarantee that GPSEE does not infringe upon any patents or copyrights, although we try very hard to insure that it does not. </p><p>There is no support for GPSEE. However, you can often find the GPSEE developers on <a href="irc://irc.freenode.org/#gpsee" rel="nofollow">irc://irc.freenode.org/#gpsee</a>, and we&#x27;re friendly. </p><h1><a name="Hot_Links"></a>Hot Links<a href="#Hot_Links" class="section_anchor"></a></h1><ul><li>Latest source tar ball (includes <a href="/p/gpsee/wiki/SpiderMonkey">SpiderMonkey</a>): <a href="http://www.page.ca/~wes/opensource/gpsee/latest-redir.html" rel="nofollow">http://www.page.ca/~wes/opensource/gpsee/latest-redir.html</a> </li><li>Building GPSEE from Sources: <a href="/p/gpsee/wiki/Building">Building</a> </li><li>Running/Writing CommonJS programs: <a href="/p/gpsee/wiki/GSR">GSR</a>, <a href="/p/gpsee/wiki/CommonJS">CommonJS</a>, Modules </li><li>Embedding GPSEE into other applications: EmbeddingGPSEE </li><li>Porting from other <a href="/p/gpsee/wiki/SpiderMonkey">SpiderMonkey</a> embeddings, see PortingToGPSEE.  </li></ul><p>   </p>
 </td>
 </tr>
</table>
<script src="http://www.gstatic.com/codesite/ph/11512667522589661435/js/prettify/prettify_core_compiled.js"></script>
<script type="text/javascript">
 prettyPrint();
</script>

 
 
 
 <script type="text/javascript" src="http://www.gstatic.com/codesite/ph/11512667522589661435/js/ph_core.js"></script>
 
 
 
 
</div> 

<div id="footer" dir="ltr">
 <div class="text">
 <a href="/projecthosting/terms.html">Terms</a> -
 <a href="http://www.google.com/privacy.html">Privacy</a> -
 <a href="/p/support/">Project Hosting Help</a>
 </div>
</div>
 <div class="hostedBy" style="margin-top: -20px;">
 <span style="vertical-align: top;">Powered by <a href="http://code.google.com/projecthosting/">Google Project Hosting</a></span>
 </div>

 
 


 
 </body>
</html>

