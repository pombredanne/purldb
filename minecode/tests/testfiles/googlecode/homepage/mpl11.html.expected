{'content_license': None,
 'content_license_url': None,
 'description': u'General Purpose Spidermonkey Embedding Ecosystem',
 'downloads_url': None,
 'external_links': {},
 'feeds_url': u'http://code.google.com/p/gpsee/feeds',
 'home_url': u'http://code.google.com/p/gpsee/',
 'id': u'gpsee',
 'issues_url': u'http://code.google.com/p/gpsee/issues/list',
 'keywords': [],
 'labels': [u'CommonJS',
            u'interoperablejs',
            u'javascript',
            u'serverjs',
            u'spidermonkey',
            u'ssjs'],
 'license': u'Mozilla Public License 1.1',
 'license_url': u'http://www.mozilla.org/MPL/',
 'name': u'gpsee',
 'peoples_url': u'http://code.google.com/p/gpsee/people/list',
 'vcs_url': u'http://code.google.com/p/gpsee/source/checkout',
 'wiki_url': u'http://code.google.com/p/gpsee/w/list'}